src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,55 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = new Object[MAX_STORAGE_SIZE];
+        this.values = new Object[MAX_STORAGE_SIZE];
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            for (int i = 0; i < size; i++) {
+                if (keys[i] == null) {
+                    values[i] = value;
+                    return;
+                }
+            }
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (key.equals(keys[i])) {
+                    values[i] = value;
+                    return;
+                }
+            }
+        }
```

Comment:
your code can be optimized into one loop, enhancing condition. think about how exactly
```suggestion
        for (int i = 0; i < size; i++) {
            if (???) {
                valueArray[i] = value;
                return;
            }
        }
```